src/Pact/Eval.hs

```diff
@@ -436,6 +467,72 @@ resolveGovernance solvedDefs m' = fmap MDModule $ forM m' $ \g -> case g of
       Nothing -> evalError (_tInfo g) "Unknown module governance reference"
     _ -> evalError (_tInfo g) "Invalid module governance, should be var"
 
+-- Auxiliary data types for strict fold on heap cost.
+data HeapMemState
+  = HeapMemState
+  { _hmMemoEnv :: !(M.Map Name Bytes)
+  , _hmTotalMem :: !Bytes
+  }
+
+data HeapFold
+  = HeapFold
+  { _hfAllDefs :: !(HM.HashMap Text Ref)
+  , _hfMemoEnv :: !(M.Map Name Bytes)
+  , _hfTotalMem :: !Bytes
+  }
+
+
+-- Inline the defuns according to the set heap limit for modules.
+-- We keep a memoized cost of each inlined `defun` so as to not calculate `sizeOf` more than once
+-- per inlined function.
+dresolveMem
+  :: Info
+  -> HeapFold
+  -> (Term (Either Text (Ref' (Term Name))), Text, c)
+  -> Eval e HeapFold
+dresolveMem info (HeapFold allDefs costMemoEnv currMem) (defTerm, defName, _) = do
+  (!unified, (HeapMemState costMemoEnv' totalMem))
+      <- runStateT (traverse (replaceMemo allDefs) defTerm)
+                 (HeapMemState costMemoEnv (sizeOf defTerm+currMem))
```

Comment:
Is this supposed to be the following?
```suggestion
                 (HeapMemState costMemoEnv ((sizeOf defTerm) + currMem))
```

Reply:
Yes, I'm sure this isn't being parsed as `sizeOf (defTerm + currmem)` but i'm :+1: on being more explicit. Will change